library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -604,7 +647,12 @@ protected void replay(long windowId)
                 pendingFiles.remove(recoveryEntry.file);
               }
               inputStream = retryFailedFile(new FailedFile(recoveryEntry.file, recoveryEntry.startOffset));
-
+              if (skipCount == 0) {
+                emitStartFileControlTuple(false);
+              }
+              if (isFailedFile) {
+                emitStartFileControlTuple(isFailedFile);
+              }
```

Comment:
If we are processing a file whose processing had failed earlier, we will emit a start file CT with a status (true) indicating that file processing had failed earlier.